event-consumer/actions/generic/index.js

```diff
@@ -0,0 +1,148 @@
+/*
+* <license header>
+*/
+
+/**
+ * This is a sample action showcasing how to access an external API
+ *
+ * Note:
+ * You might want to disable authentication and authorization checks against Adobe Identity Management System for a generic action. In that case:
+ *   - Remove the require-adobe-auth annotation for this action in the manifest.yml of your application
+ *   - Remove the Authorization header from the array passed in checkMissingRequestInputs
+ *   - The two steps above imply that every client knowing the URL to this deployed action will be able to invoke it without any authentication and authorization checks against Adobe Identity Management System
+ *   - Make sure to validate these changes against your security requirements before deploying the action
+ */
+
+const fetch = require('node-fetch')
+const { Core, Events } = require('@adobe/aio-sdk')
+const { context, getToken } = require('@adobe/aio-lib-ims')
```

Comment:
Do you have to use `aio-lib-ims` to generate the token or is there another way ? I'm asking because in the README https://github.com/adobe/aio-lib-ims#running-in-an-adobe-io-runtime-action it says: 

> Note that Project Firefly applications should not own the responsibility to generate their own IMS access tokens. We strongly discourage this approach in favor of more secure implementation patterns that are documented in our Project Firefly Security Guide.

Reply:
AFAIK the real meaning of this note is: please don't use aio-lib-ims when you don't absolutely need to. In this case, you have to use it, and the security risk is minimal because the action runs itself by alarms, no external user is allowed to touch it (not a web action).